### ISSUE TITLE
Cannot remove all recommended themes using django admin

### ISSUE THREAD

**Issue by ioanarusiczki** [mozilla/addons-server#11825]
Followup for https://github.com/mozilla/addons-server/issues/10403
STR:
1. Load https://addons-internal.dev.mozaws.net/en-US/admin/models/bandwagon/collection/2143965/change/
2. Remove all themes from this collection and save changes
3. Load AMO dev 

Expected result:
`Recommended themes` shelf is no longer available on the hompeage.

Actual result:
Shelf is displayed with 3 themes and See more recommended themes can be accessed 
https://addons-dev.allizom.org/en-US/firefox/search/?recommended=true&sort=random&type=statictheme

Notes:
Reproducible on dev with FF68(Win10).

This is the frontend 
![frontend for themes](https://user-images.githubusercontent.com/33448286/61139331-cf076800-a4d1-11e9-8d59-a795e31acf6a.png)

The collection is empty in admin
![no themes in admin](https://user-images.githubusercontent.com/33448286/61139341-d464b280-a4d1-11e9-8821-361606a8e43e.png)

**Comment by eviljeff** [comment]
Its just if you remove all of them, right? Removing one or more but leaving at least one works correctly?

**Comment by ioanarusiczki** [comment]
@eviljeff Sorry for the late answer, I've been away. 

I checked this again with FF68 (win10) and I could reproduce it on dev : adding and removing works but when all themes are removed and django shows no content in this collection the frontend will continue to display the following 3 themes:

504236 Underwater Dolphins
616881 Bubble mode 7
498533 Flag of Colombia 

If I leave one the frontend will display 4 themes (what's left in django plus those 3 themes). 

But I could not reproduce it on AMO stage. If I remove all themes the frontend shows it correctly.

stage 
![stage](https://user-images.githubusercontent.com/33448286/62032346-15f39e00-b1f2-11e9-8ecf-0c8d99828aca.png)

dev 
![dev](https://user-images.githubusercontent.com/33448286/62032361-20159c80-b1f2-11e9-8b70-a3dc00352472.png)

**Comment by eviljeff** [comment]
if the themes are still showing as recommended in the frontend (and they are in the api response too), but aren't in the featured themes collection, then it's possible (probable, imo) they've been directly added as a recommendable add-ons and then approved - i.e. the have DiscoveryItem entries that have been marked as `recommendable` and then the next version approved)

**Comment by ioanarusiczki** [comment]
@eviljeff I managed to delete successfully 498533 Flag of Colombia. It was added like you said, in the Discovery items entries (like the other [two](https://github.com/mozilla/addons-server/issues/11825#issuecomment-515892850) mentioned above). In it's place I added with the theme collection 497656 Colombia. 

But when I tried to do the same thing for the other two themes - delete them from the Discovery items and add them back using the collection, it did not work anymore. 
These two themes: 
504236 Underwater Dolphins
616881 Bubble mode 7 
still show up in the search API as being recommended. (yet, they are deleted from discovery items).

https://addons-dev.allizom.org/api/v4/addons/search/?recommended=true&type=statictheme

Yet, the badge is no longer there on the detail page. Also when I checked the detail API for these two themes, I see `is_recommended:false`. 

https://addons-dev.allizom.org/api/v4/addons/addon/504236
https://addons-dev.allizom.org/api/v4/addons/addon/616881
